supabase/migrations/22_inferred_schemas.sql

```diff
@@ -0,0 +1,28 @@
+begin;
+
+create table inferred_schemas (
+    collection_name  catalog_name not null,
+    "schema"         json not null,
+    flow_document    json not null,
+    primary key (collection_name)
+);
+alter table inferred_schemas enable row level security;
+
+create policy "Users must be authorized to the collection name"
+  on inferred_schemas as permissive for select
+  using (exists(
+    select 1 from auth_roles('read') r where collection_name ^@ r.role_prefix
```

Comment:
Just to make sure, `auth_roles` is  the function that unrolls all of your direct and indirect grants, right? As in, this isn't checking just the collections you have direct grants to, but also the ones that your tenant has grants to, right?

Reply:
Correct. This returns anything you can access with a `read` capability (or above)